src/services/user/firstLogin.js

```diff
@@ -75,6 +76,30 @@ const firstLogin = async (data, params, app) => {
 			termsOfUseConsent: data.termsOfUseConsent,
 		};
 	}
+
+	if (data.termsOfUseConsentVersion || data.privacyConsentVersion) {
+		updateConsentUsingVersions = app.service('consents').find({ userId: user._id }).then((consents) => {
+			if (consents.total === 0) {
+				throw new Error('user consent not found!');
+			}
+			const consent = consents.data[0];
+			const consentVersionUpdate = {
+				userConsent: {
+					form: 'digital',
+				},
+			};
+			if (data.privacyConsentVersion) {
+				consentVersionUpdate.userConsent.privacyConsent = true;
+				consentVersionUpdate.userConsent.dateOfPrivacyConsent = Date.now();
+			}
+			if (data.termsOfUseConsentVersion) {
+				consentVersionUpdate.userConsent.termsOfUseConsent = true;
+				consentVersionUpdate.userConsent.dateOfTermsOfUseConsent = Date.now();
+			}
+			return app.service('consents').patch(consent._id, { $set: consentVersionUpdate });
```

Comment:
@Metauriel how to update the termsOfUseConsent & -date or the privacyConsent only here using set by using the hook for setting the related date while keeping the other consent data and not overwriting the date fields when having the whole consent sent to the service?